public/app/plugins/datasource/influxdb/response_parser.ts

```diff
@@ -50,6 +53,106 @@ export default class ResponseParser {
     // order is insertion-order, so this should be ok.
     return Array.from(res).map((v) => ({ text: v }));
   }
+
+  getTable(dfs: DataFrame[], target: InfluxQuery, meta: QueryResultMeta): TableModel {
+    let table = new TableModel();
+
+    if (dfs.length > 0) {
+      table.meta = {
+        ...meta,
+        executedQueryString: dfs[0].meta?.executedQueryString,
+      };
+
+      table.refId = target.refId;
+      table = getTableCols(dfs, table, target);
+
+      // if group by tag(s) added
+      if (dfs[0].fields[1].labels) {
+        let dfsByLabels: any = groupBy(dfs, (df: DataFrame) =>
+          df.fields[1].labels ? Object.values(df.fields[1].labels!) : null
+        );
+        const labels = Object.keys(dfsByLabels);
+        dfsByLabels = Object.values(dfsByLabels);
+
+        for (let i = 0; i < dfsByLabels.length; i++) {
+          table = getTableRows(dfsByLabels[i], table, [...labels[i].split(',')]);
+        }
```

Comment:
(this can be done after the PR is merged)

i think this part could be simplified with using `Object.entries`, something like this.
```suggestion
        const dfsByLabels = groupBy(dfs, (df: DataFrame) =>
          df.fields[1].labels ? Object.values(df.fields[1].labels!) : null
        );

        Object.entries(dfsByLabels).forEach(([key,value]) => {
          table = getTableRows(value, table, [...key.split(',')]);
        });
```